website/static/js/pages/project-settings-page.js

```diff
@@ -128,25 +128,77 @@ $(document).ready(function() {
-                } else {
-                    window.location.reload();
-                }
+        
+        var submit = function() {
+            var request = $osf.postJSON(ctx.node.urls.api + 'settings/addons/', formData);
+            return request;
+            
+        };
+        function successfulAddonUpdate() {
+            msgElm.text('Settings updated').fadeIn();
+            checkedOnLoad = $('#selectAddonsForm input:checked');
+            uncheckedOnLoad = $('#selectAddonsForm input:not(:checked)');
+            if($osf.isSafari()) {
+                        //Safari can't update jquery style change before reloading. So delay is applied here
+                        setTimeout(function(){window.location.reload();}, 100);
+            } else {
+                        window.location.reload();
             }
-        });
+        }
+        function failedAddonUpdate() {
+            var msg = 'Sorry, we had trouble saving your settings. If this persists please contact <a href="mailto: support@osf.io">support@osf.io</a>';
+            bootbox.alert({
+                title: 'Request failed',
+                message: msg,
+                buttons: {
+                    ok: {
+                        label: 'Close',
+                        className: 'btn-default'
+                    }
+                }
+            });
+        }
+        // some addons disabled (unchecked warning adopted from profile-settings-addons-page.js)
+        if(unchecked.length > 0) {
+            var uncheckedText = $.map(unchecked, function(el){
+                return ['<li>', $(el).closest('label').text().trim(), '</li>'].join('');
+            }).join('');
+            uncheckedText = ['<ul>', uncheckedText, '</ul>'].join('');
+            bootbox.confirm({
+                title: 'Are you sure you want to remove the add-ons you have deselected? ',
+                message: uncheckedText,
+                callback: function(result) {
+                    if (result) {
+                        var request = submit();
+                        request.done(successfulAddonUpdate);
+                        request.fail(failedAddonUpdate);
+                    } else{
+                        unchecked.each(function(i, el){ $(el).prop('checked', true); });
+                    }
+                },
+                buttons: {
+                    confirm: {
+                        label: 'Remove',
+                        className: 'btn-danger'
+                    }
+                }
+            });
+        }
+        //no addons disabled but some addons enabled
+        else if(checked.length>0) {
+            var request = submit();
+            request.done(successfulAddonUpdate);
+            request.fail(failedAddonUpdate);
+        }
+        // no changes to the state of the addons
+        else {
+            msgElm.text('Settings updated').fadeIn();
+            msgElm.fadeOut(800);
```

Comment:
I didn't give a verdict on this but I agree with @abought's thoughts on this. Consistency is important in success messages. There is nothing wrong with showing success message even if nothing is saved. It doesn't help the user if we say "Saving worked but you actually didn't change anything". 

For the style guide we need to think about whether to stop save button from being available until actual changes take place but this step is unnecessary. I think the closest thing we have right now is in error reporting section "Expected but critical success event" https://centerforopenscience.github.io/osf-style/ui.html#error-reporting